### DISCUSSION TITLE
Multiple pdf file upload using axios post api call is not working with fastapi: error response (422: Unprocessable entity)

### DISCUSSION THREAD

**Discussion by jemis140** [fastapi/fastapi#10160] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
n/A
```

### Description

I am currently working on NLP project and I am using fastapi as a backend API and React +  Ant Design in the front. 

Problem: I am currently working on a feature where users upload multiple PDF files to get a pickle file which combines Vector embeddings of PDF files as the response to an API call. I am getting error 422: Unprocessable entity in a frontend API call. I am using Axios for requests. The following is the relevant code snippets and screenshot of the network tabs.

**Fast API route:**

```
@router.post("/getconversationchain")
async def get_conversation_chain(files: List[UploadFile] = File(...)):
```

**Swagger UI API test screenshot:**

<img width="648" alt="Screen Shot 2023-08-27 at 7 12 37 PM" src="https://github.com/tiangolo/fastapi/assets/28834808/b9189335-5d3d-4498-83b5-5fae8e23ca81">

**PDFapi.js file relevant code snippet** 

```
export const getConversationChain = async (files) => {
  try {
    const formData = new FormData();

    files.forEach((file) => {
      formData.append("file", file.originFileObj);
    });

    const response = await axios.post(
      `${BASE_URL}/getconversationchain`,
      formData
    );

    return response.data;
  } catch (error) {
    throw error;
  }
};
```

**Chrome browser network tab screenshots:**

 **1) I am getting files object and the following is the screenshot of console.log.**
<img width="403" alt="Screen Shot 2023-08-27 at 7 14 49 PM" src="https://github.com/tiangolo/fastapi/assets/28834808/beac933b-40fb-4691-b48f-41d163495fee">

**2) Headers of request**
<img width="631" alt="Screen Shot 2023-08-27 at 7 09 35 PM" src="https://github.com/tiangolo/fastapi/assets/28834808/238e4d3e-2acc-4b2d-9f00-6cfce20ffdd7">

**3) Payload of request: I uploaded two PDF files and I guess there is no issue with the payload.**
<img width="786" alt="Screen Shot 2023-08-27 at 7 09 44 PM" src="https://github.com/tiangolo/fastapi/assets/28834808/ea12a670-8aed-48b8-ac2e-455e4266065d">

**4) Response from API:**
<img width="482" alt="Screen Shot 2023-08-27 at 7 09 54 PM" src="https://github.com/tiangolo/fastapi/assets/28834808/358a4010-3cf4-4e38-b82f-5670a8ddacd2">

The following is the discussion I found on the closed issue on this repo and didn't find any satisfactory answer to what is happening under the hood and how multi-part/form-data requests actually work in FastAPI. I also tried quick fixes from removing the content header and changing the params of the route from List[UploadFile] to Annoted[bytes]. also, converted files into Bolb objects to implement a solution for this problem. I have exhausted every option I can think of. If somebody knows a way around please guide me here. Thank you.

Similar question : [File upload difficulties](https://github.com/tiangolo/fastapi/discussions/7074)

### Operating System

macOS

### FastAPI Version

0.101.1

### Pydantic Version

1.10.12

### Python Version

3.9

**Comment by chrisK824** [comment, marked as answer]
Are you sure you are trying with the correct keyword in your form data? Swagger documentation suggests that you expect for keyword "files" whereas in axios call you are using the f=keyword "file". Maybe it was a typo all along?

**Comment by jemis140** [comment]
@chrisK824  thanks for responding. yeah, I missed one character "files". I realized it later that day when I posted this quesiton. thanks though for your keen observation. I am closing this discussion.

**Comment by jemis140** [comment]
Error resolved.
